python/cudf/cudf/core/dtypes.py

```diff
@@ -166,3 +163,51 @@ def __repr__(self):
             return f"ListDtype({self.element_type.__repr__()})"
         else:
             return f"ListDtype({self.element_type})"
+
+
+class StructDtype(ExtensionDtype):
+
+    name = "struct"
+
+    def __init__(self, fields):
+        """
+        fields : dict
+            A mapping of field names to dtypes
+        """
+        pa_fields = {
+            k: cudf.utils.dtypes.cudf_dtype_to_pa_type(v)
+            for k, v in fields.items()
+        }
+        self._typ = pa.struct(pa_fields)
+
+    @property
+    def fields(self):
+        return {
+            field.name: cudf.utils.dtypes.cudf_dtype_from_pa_type(field.type)
+            for field in self._typ
+        }
```

Comment:
Just a general comment: py3.7+ starts to make "insertion order" to `dict` official ([ref](https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6)). So this should make if `lhs==rhs`, then `lhs.fields==rhs.fields`.

Maybe add a test case for this too?

Reply:
Yes, StructDtype fields are ordere, so we're implicitly dependent on this behaviour. Added tests for the `.fields` method.